pkg/api/plugin.go

```diff
@@ -14,12 +14,19 @@ const (
 	ContextKeyTenantID     ContextKey = "TenantID"
 )
 
+// DeployFn makes it possible to plug in different logic to the deploy.
+// this should call:
+// - clients.Deployments.CreateOrUpdate() to deploy the arm
+type DeployFn func(context.Context, *OpenShiftManagedCluster, []byte, PluginConfig) error
```

Comment:
Not sure DeployFn should need to know about PluginConfig.

Reply:
I need it at the moment to get at the acceptLanguages for the clients, but we *could* do without that.

Reply:
I am going to move the commits in this PR and add them to https://github.com/openshift/openshift-azure/pull/521 as Jim asked. I'll fold your requests into that.